Fundamentals/Backoffice/Property-Editors/Built-in-Property-Editors/Block-Editor/build-custom-view-for-blocks.md

```diff
@@ -0,0 +1,133 @@
+---
+versionFrom: 9.0.0
+versionTo: 10.0.0
+---
+
+# Build a Custom View for a Block
+
+You can choose to customize your editing experience by implementing a custom view for each Block Type of a Block Editor.
+
+By picking a custom view you overwrite the backoffice UI for the given block with your own. This enables you to define how a block should be presented. It can, however, also include interactive elements and be a full custom solution to how data is manipulated. In the Block List Editor the Inline Editing Mode must be disabled for custom backoffice views to appear.
+
+## Write your own HTML view
+
+Currently you can only pick HTML files for a custom view. These views are powered by AngularJS and you can write any AngularJS logic.
+
+Your HTML can be anything. If you use it as a representation of the content you would also want the full view to be clickable. This would then open the default editor for editing of your content.
+
+The following example displays the property with the alias `headline` together with the `description` inside a button to edit your block.
+
+```html
+<button type="button" ng-click="block.edit()">
+  <h2 ng-bind="block.data.headline"></h2>
+  <p ng-bind="block.data.description"></p>
+</button>
+```
+
+If you would like to display properties of `settings`, you can access these by using `block.settingsData.myPropertyAlias`.
+
+## Add the Areas Container for Block Grid Editor
+
+Blocks of the Block Grid Editor can have Areas. These Blocks requires their Custom View to define where the Area Container should be inserted.
+
+This enables you to wrap and place the Areas as you need.
+
+The Area Container is inserted through a slot. Insert the following snippet where you would like the Area to appear in your Custom View:
+
+```html
+<slot name="areas" part="areas"></slot>
+```
+
+## Adding custom implementation to your View
+
+To achieve this you need to add a custom AngularJS controller to your custom view, using the `ng-controller` attribute:
+
+```html
+<button type="button" ng-controller="customBlockController" ng-click="api.editBlock(block, block.hideContentInOverlay, index, parentForm)" class="btn-reset umb-outline blockelement-labelblock-editor blockelement__draggable-element ng-scope ui-sortable-handle" >
```

Comment:
@nielslyngsoe I am not sure we need all these CSS classes do we?
I know for sure `ng-scope` is not needed